x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateDiff.java

```diff
@@ -166,6 +173,40 @@ public DateDiff(
         this.endTimestamp = endTimestamp;
     }
 
+    private DateDiff(StreamInput in) throws IOException {
+        this(
+            Source.readFrom((PlanStreamInput) in),
+            ((PlanStreamInput) in).readExpression(),
+            ((PlanStreamInput) in).readExpression(),
+            ((PlanStreamInput) in).readExpression()
+        );
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        Source.EMPTY.writeTo(out);
+        ((PlanStreamOutput) out).writeExpression(unit);
+        ((PlanStreamOutput) out).writeExpression(startTimestamp);
+        ((PlanStreamOutput) out).writeExpression(endTimestamp);
```

Comment:
All of these'll get replaced with `writeNamedWriteable` soon. We'll still have the source cast, but only that one.